src/java/org/apache/cassandra/index/sai/disk/format/IndexComponent.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.format;
+
+/**
+ * This is a definitive list of all the on-disk components for all versions
+ */
+public enum IndexComponent
+{
+    /**
+     * Stores per-index metadata.
+     *
+     * V1
+     */
+    META("Meta"),
+    /**
+     * Term dictionary written by {@code TrieTermsDictionaryWriter} stores mappings of term and
+     * file pointer to posting block on posting file.
+     *
+     * V1
+     */
+    TERMS_DATA("TermsData"),
+    /**
+     * Stores postings written by {@code PostingsWriter}
```

Comment:
```suggestion
     * Stores postings written by {@link PostingsWriter}
```
...but it might make us add an import